packages/client/src/v2-events/messages/utils.tsx

```diff
@@ -191,3 +194,26 @@ export function useIntlFormatMessageWithFlattenedParams() {
       variablesUsed(getDefaultMessage(message))
   }
 }
+
+export function useIntlWithFormData() {
```

Comment:
I'm considering if this should actually be built in to the existing `useIntlFormatMessageWithFlattenedParams()` so that we could always just use that single function for formatting messages with params/data. But this is fine for now, we can change it later if necessary.